src/app/services/api/api.service.ts

```diff
@@ -18,61 +18,117 @@ import { Blockchain, GetBlocksResponse, GetBlockchainMetadataResponse, GetUnconf
  */
 @Injectable()
 export class ApiService {
+  /**
+   * Key used for saving the local node URL in the persistent storage and the window object.
+   */
+  private static readonly localNodeUrlKey = 'nodeUrl';
 
   /**
    * URL used to connect to the API.
    */
   private url = '/api/';
+  /**
+   * Subject for emitting every time the local node URL is changed or removed.
+   */
+  private localNodeUrlSubject: ReplaySubject<string> = new ReplaySubject<string>(1);
 
   constructor(
     private http: HttpClient
   ) { }
 
+  /**
+   * Initializes the service.
+   */
+  initialize() {
+    // Get the URL of the local node that must be used as backend, if an URL was saved before.
+    const localNodeUrl = localStorage.getItem(ApiService.localNodeUrlKey);
+    if (localNodeUrl) {
+      window[ApiService.localNodeUrlKey] = localNodeUrl;
+    }
+
+    this.localNodeUrlSubject.next(localNodeUrl);
+  }
+
+  /**
+   * Sets the URL of the local node that must be used as backend. If the URL is null, any
+   * previously saved URL is removed.
+   */
+  setNodeUrl(url: string) {
+    window[ApiService.localNodeUrlKey] = url;
+
+    if (url) {
+      localStorage.setItem(ApiService.localNodeUrlKey, url);
+    } else {
+      localStorage.removeItem(ApiService.localNodeUrlKey);
+    }
+
+    this.localNodeUrlSubject.next(url);
+  }
+
+  /**
+   * Emits every time the local node URL is changed or removed.
+   */
+  get localNodeUrl(): Observable<string> {
+    return this.localNodeUrlSubject.asObservable();
+  }
+
   /**
    * Get information about the state of the node.
    */
   getHealth(): Observable<any> {
-    return this.get('health');
+    const url = !this.nodeUrl() ? 'health' : 'v1/health';
+
+    return this.get(url);
   }
 
   /**
    * Gets the list of transactions of a specific address.
    * @param address Address to consult.
    */
   getAddress(address: string): Observable<GetTransactionResponse[]> {
-    return this.get('transactions', { addrs: address, verbose: 1 });
+    const url = !this.nodeUrl() ? 'transactions' : 'v1/transactions';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin-explorer/pullRequest?prid=7163626)
```suggestion
    const url = !this.nodeUrl() ? "transactions" : 'v1/transactions';
```